### DISCUSSION TITLE
Customize arc link labels

### DISCUSSION THREAD

**Discussion by johannesKnoll** [plouc/nivo#1609] [Q&A] [answered]
Hi all,

I am currently trying to customize the arc link labels for my Donut Chart. Are there any possibilities to return some HTML-Tags for the labels, e.g buttons or headings?

Many thanks!

Cheers,
Johannes

**Comment by wyze** [comment, marked as answer]
You can customize the component rendered but it still must return valid SVG elements.

**Reply by johannesKnoll** [reply]
Hi,
thanks for your reply. I tried to return SVG elements, but that didn't work either.
Is there another possibility to render other elements expect plain text?

Thanks,
Johannes

**Reply by wyze** [reply]
Here is an example of the arc labels component being customized: https://nivo.rocks/storybook/?path=/story/pie--custom-arc-label-component and here is the source code that makes that happen: https://github.com/plouc/nivo/blob/c2a656e234e41646c16e2e8ba63dead811f804f7/packages/pie/stories/pie.stories.tsx#L207-L223 .. You will just change the `arcLabelsComponent` prop to `arcLinkLabelComponent` and it will allow you to have full control over the component.

And here is the default implementation: https://github.com/plouc/nivo/blob/c2a656e234e41646c16e2e8ba63dead811f804f7/packages/arcs/src/arc_link_labels/ArcLinkLabel.tsx

**Reply by johannesKnoll** [reply]
Thank you very much! That worked fine for me

**Reply by chhakuli123** [reply]
@johannesKnoll 
Please help me with how we can customize arc arc-link label in our own way. My requirement is like this 
![image](https://github.com/plouc/nivo/assets/103095813/954ef6a1-6ceb-4257-a323-8e009b9c4ae4)

**Reply by chhakuli123** [reply]
@wyze If you can help in making these arc link labels.

**Reply by adityachat360** [reply]
@chhakuli123  did you get the solution for custom arc arc-link label

**Reply by chhakuli123** [reply]
@adityachat360 No sorry
